GetandTake/Areas/Identity/Pages/Account/ExternalLogin.cshtml.cs

```diff
@@ -0,0 +1,197 @@
+    }
+    
+    public IActionResult OnGet() => RedirectToPage("./Login");
+
+    public IActionResult OnPost(string provider, string returnUrl = null)
+    {
+        var redirectUrl = Url.Page("./ExternalLogin", pageHandler: "Callback", values: new { returnUrl });
+        var properties = _signInManager.ConfigureExternalAuthenticationProperties(provider, redirectUrl);
+        
+        return new ChallengeResult(provider, properties);
+    }
+
+    public async Task<IActionResult> OnGetCallbackAsync(string returnUrl = null, string remoteError = null)
+    {
+        returnUrl = returnUrl ?? Url.Content("~/");
+        
+        if (remoteError != null)
+        {
+            ErrorMessage = $"Error from external provider: {remoteError}";
+            
+            return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+        }
+        var info = await _signInManager.GetExternalLoginInfoAsync();
+        
+        if (info == null)
+        {
+            ErrorMessage = "Error loading external login information.";
+            
+            return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+        }
+
+        var result = await _signInManager.ExternalLoginSignInAsync(info.LoginProvider, info.ProviderKey, isPersistent: false, bypassTwoFactor: true);
+        
+        if (result.Succeeded)
+        {
+            _logger.LogInformation("{Name} logged in with {LoginProvider} provider.", info.Principal.Identity.Name, info.LoginProvider);
+            
+            return LocalRedirect(returnUrl);
+        }
+        if (result.IsLockedOut)
+        {
+            return RedirectToPage("./Lockout");
+        }
+        else
+        {
+            ReturnUrl = returnUrl;
+            ProviderDisplayName = info.ProviderDisplayName;
+            
+            if (info.Principal.HasClaim(c => c.Type == ClaimTypes.Email))
+            {
+                Input = new InputModel
+                {
+                    Email = info.Principal.FindFirstValue(ClaimTypes.Email)
+                };
+            }
+            return Page();
+        }
+    }
+
+    public async Task<IActionResult> OnPostConfirmationAsync(string returnUrl = null)
+    {
+        returnUrl = returnUrl ?? Url.Content("~/");
+        var info = await _signInManager.GetExternalLoginInfoAsync();
+        
+        if (info == null)
+        {
+            ErrorMessage = "Error loading external login information during confirmation.";
+            
+            return RedirectToPage("./Login", new { ReturnUrl = returnUrl });
+        }
+
+        if (ModelState.IsValid)
+        {
+            var user = CreateUser();
+
+            await _userStore.SetUserNameAsync(user, Input.Email, CancellationToken.None);
+            await _emailStore.SetEmailAsync(user, Input.Email, CancellationToken.None);
+
+            var result = await _userManager.CreateAsync(user);
+            
+            if (result.Succeeded)
+            {
```

Comment:
Let's refactor these huge `if` nesting, it makes code really difficult to read and maintain